### PR TITLE
1421 report dates in url

### PR DESCRIPTION
**Pull request by Alastair-L**
### Issue #:
https://github.com/beyondessential/tupaia-backlog/issues/1421

I have decided to leave all dates in `day` form, like so: `reportPeriod=1st_Sep_2020-30th_Sep_2020`
This has 2 advantages:
1. It better represents the actual state.
2. It can be interpreted without knowing the periodGranularity.

I think possibly the plan should be to pass start and end dates around in the string format? i.e. `W12020`. Either way, I don't think it's in the scope of this PR?

### CODE REVIEW

File: packages/web-frontend/src/historyNavigation/constants.js [hunk 1]
```diff
@@ -14,14 +14,9 @@ const DASHBOARD = 'DASHBOARD';
 const MEASURE = 'MEASURE';
 const REPORT = 'REPORT';
 const PASSWORD_RESET_TOKEN = 'PASSWORD_RESET_TOKEN';
-const TIMEZONE = 'TIMEZONE';
-const START_DATE = 'START_DATE';
-const END_DATE = 'END_DATE';
-const DISASTER_START_DATE = 'DISASTER_START_DATE';
-const DISASTER_END_DATE = 'DISASTER_END_DATE';
```
Comment (kael89) [inline packages/web-frontend/src/historyNavigation/constants.js:21]: Yay for all the simplification!

File: packages/web-frontend/src/selectors/dashboardSelectors.js [hunk 2]
```diff
@@ -15,6 +20,11 @@ export const selectCurrentExpandedViewId = createSelector([selectLocation], loca
   getLocationComponentValue(location, URL_COMPONENTS.REPORT),
 );
 
+export const selectCurrentExpandedDates = createSelector([selectLocation], location => {
+  const startDateString = getLocationComponentValue(location, URL_COMPONENTS.REPORT_PERIOD) ?? '';
+  return convertUrlPeriodStringToDateRange(startDateString);
```
Comment (kael89) [inline packages/web-frontend/src/selectors/dashboardSelectors.js:25]:
```suggestion
  const periodString = getLocationComponentValue(location, URL_COMPONENTS.REPORT_PERIOD) ?? '';
  return convertUrlPeriodStringToDateRange(periodString);
```

File: packages/web-frontend/src/containers/EnlargedDialog/EnlargedDialogContent.js [hunk 3]
```diff
@@ -219,6 +219,10 @@ export class EnlargedDialogContent extends PureComponent {
     const { onSetDateRange, isLoading, viewContent, isExporting } = this.props;
     const { periodGranularity, startDate, endDate } = viewContent;
 
+    if (!periodGranularity) {
+      return null;
+    }
```
Comment (kael89) [inline packages/web-frontend/src/containers/EnlargedDialog/EnlargedDialogContent.js:224]: Question: this makes a lot of sense, how come it wasn't an issue in the past? Or was it (does it fix an existing bug?)

**Review by kael89 — CHANGES_REQUESTED** [review]
Awesome work @Alastair-L !! It's really exiting seeing this is almost ready to go, it will be a very useful functionality 🚀

Only a small renaming requested + a question

File: packages/web-frontend/src/historyNavigation/constants.js (hunk 1 above)
Comment (Alastair-L) [reply packages/web-frontend/src/historyNavigation/constants.js:21]: Tom's new export system is amazing

File: packages/web-frontend/src/selectors/dashboardSelectors.js (hunk 2 above)
Comment (Alastair-L) [reply packages/web-frontend/src/selectors/dashboardSelectors.js:25]: Oops! Good catch

File: packages/web-frontend/src/containers/EnlargedDialog/EnlargedDialogContent.js (hunk 3 above)
Comment (Alastair-L) [reply packages/web-frontend/src/containers/EnlargedDialog/EnlargedDialogContent.js]: I believe that this is completely unneeded! The only place this is rendered is in this file (line 122) and states:
`{periodGranularity && this.renderPeriodSelector()}`

**Review by kael89 — APPROVED** [review]
Looks great!

**Pull request merged**